activerecord/test/cases/tasks/legacy_database_tasks_test.rb

```diff
@@ -0,0 +1,1005 @@
+# frozen_string_literal: true
+
+require "cases/helper"
+require "active_record/tasks/database_tasks"
+
+module ActiveRecord
+  class LegacyDatabaseTasksUtilsTask < ActiveRecord::TestCase
+    def test_raises_an_error_when_called_with_protected_environment
+      ActiveRecord::MigrationContext.any_instance.stubs(:current_version).returns(1)
```

Comment:
We need to use `assert_called_on_instance_of` since depending on `mocha` has been removed at #33162.